cartography/intel/crxcavator/crxcavator.py

```diff
@@ -49,55 +52,139 @@ def call_crxcavator_api(api_and_parameters, crxcavator_api_key, crxcavator_base_
-                'privacy_policy': data['webstore'].get('privacy_policy'),
-                'rating': data['webstore'].get('rating'),
-                'rating_users': data['webstore'].get('rating_users'),
-                'short_description': data['webstore'].get('short_description'),
-                'size': data['webstore'].get('size'),
-                'support_site': data['webstore'].get('support_site'),
-                'users': data['webstore'].get('users'),
-                'website': data['webstore'].get('website'),
-                'type': data['webstore'].get('type'),
-                'price': data['webstore'].get('price'),
-                'report_link': "https://crxcavator.io/report/" + extension_id + "/" + version,
-            })
+    for extension in extension_details:
+        extension_id = extension['extension_id']
+        version = extension['version']
+        data = extension.get('data')
+        if not data:
+            logger.warning(f'Could not retrieve details for extension {extension}')
+            continue
+        risk_data = data.get('risk')
+        if not risk_data:
+            risk = {}
+        else:
+            risk = parse_risk(risk_data)
+        webstore = data.get('webstore', {})
+        extensions.append({
+            'id': f"{extension_id}|{version}",
+            'extension_id': extension_id,
+            'version': version,
+            'risk_total': risk.get('total'),
+            'risk_permissions_score': risk.get('permissions_score'),
+            'risk_webstore_score': risk.get('webstore_score'),
+            'risk_metadata': risk.get('metadata'),
+            'risk_optional_permissions_score': risk.get('optional_permissions_score'),
+            'risk_csp_score': risk.get('csp_score'),
+            'risk_extcalls_score': risk.get('extcalls_score'),
+            'risk_vuln_score': risk.get('vuln_score'),
+            'address': webstore.get('address'),
+            'email': webstore.get('email'),
+            'icon': webstore.get('icon'),
+            'crxcavator_last_updated': webstore.get('last_updated'),
+            'name': webstore.get('name'),
+            'offered_by': webstore.get('offered_by'),
+            'permissions_warnings': webstore.get('permission_warnings'),
+            'privacy_policy': webstore.get('privacy_policy'),
+            'rating': webstore.get('rating'),
+            'rating_users': webstore.get('rating_users'),
+            'short_description': webstore.get('short_description'),
+            'size': webstore.get('size'),
+            'support_site': webstore.get('support_site'),
+            'users': webstore.get('users'),
+            'website': webstore.get('website'),
+            'type': webstore.get('type'),
+            'price': webstore.get('price'),
+            'report_link': f"https://crxcavator.io/report/{extension_id}/{version}",
+        })
     return extensions
 
 
+def parse_risk(risk_data):
```

Comment:
This function can probably be refactored to be a bit more general.

```
def get_data(data_dict, key):
    data = data_dict.get(key)
    data_score = 0
    if data:
        data_score = permissions.get('total')
    return data, data_score
```

Thus you'd be able to just call a bunch of lines with:

```
permissions, permissions_score = get_data(risk_data, 'permissions')
...
```